### ISSUE TITLE
how to change  m

### ISSUE THREAD

**Issue by zjz5250** [wy1iu/sphereface#78]
at the beginning，i use m  as  “type：SINGLE”，and  the loss converge，acc can is about  0.98.
but  when i  change the  m as “type：QUADRUPLE”，the  loss become  larger and larger，and acc come down also。
  so  how to change  the type  correctly？
   hope  the  help

**Comment by zuoqing1988** [comment]
Directly using QUADRUPLE is OK for CASIA.
But for other datasets (for example, subsets of ms_celeb_1m), it is hard to converge.
When finetuning QUADRUPLE from the result of SINGLE, set lr<=0.01 and base = 10, lambda_min = 10.
Try many times, it may converges.

**Comment by zjz5250** [comment]
thanks

**Comment by zjz5250** [comment]
@zuoqing1988
 yes，when i set lr and base as you said，it worked，the loss began to converge obviously。
then，could  you tell me  the rule  about  how to change  the lr  and base？when the type changed。
thx

**Comment by zuoqing1988** [comment]
@zjz5250 Finetuning QUADRUPLE from SINGLE, lr = 0.01 base = 10, lambda_min=10, will be OK in very high probability. But if it diverges, try smaller lr.
When QUADRUPLE converges with some lambda_min (10), You can also finetune with smaller lambda_min (5, 2 or 1). You can try argument "-snapshot xxx_iter_xxx.caffemodel"。

**Comment by zuoqing1988** [comment]
@zjz5250 As far as I know, nobody in this forum successes to train QUADRUPLE on such a big dataset.

**Comment by zjz5250** [comment]
@zuoqing1988 
  so  if i train  SINGLE 、DOUBLE OR TRIPLE，it can converge？ 
and  how to set the  lr 、 base 、 lambda_min

**Comment by zuoqing1988** [comment]
@zjz5250 I have trained on cleaned subsets of MS-Celeb-1M, around 80,000 persons. SINGLE is easy to converge, but acc on LFW is less than 99%. TRIPLE and QUADRUPLE are hard to converge.
https://github.com/wy1iu/sphereface/issues/14
#14 
The largest dataset I have successed to train with QUADRUPLE includes around 30,000 person, 2 millon images.

**Comment by zjz5250** [comment]
@zuoqing1988
  sorry，i made   a   mistake。
  i mean  that  i want to  train it  with a dataset  about  50 000 person，110 000 pics。
  i set  base_lr=0.001   
   momentum: 0.9
  lr_policy: "multistep"
  stepvalue: 32000
  stepvalue: 48000
  stepvalue: 60000
  gamma: 0.1
  weight_decay:0.0005
and:
   base: 1000
   gamma: 0.12
   power: 1
   lambda_min: 10
   iteration: 0
but  even SINGLE  can not  not  converge.
  can you tell me  your details  about  these  parameters，when trained   on cleaned subsets of MS-Celeb-1M

**Comment by zuoqing1988** [comment]
@zjz5250 
I have select a subset of MS_Celeb_1M with 45,971 people,  3645724 images.
It converges very fast for SINGLE (initial loss = 10.7, after 10,000 iters loss = 7.0,  after 20,000 iters loss is less than 1.5).
base_lr: 0.01
lr_policy: "multistep"
gamma: 0.1

stepvalue: 160000
stepvalue: 240000
max_iter: 50000

display: 100
momentum: 0.9
weight_decay: 0.0005

base: 1000
gamma: 0.08
power: 1
lambda_min: 5
iteration: 0

**Comment by zjz5250** [comment]
@zuoqing1988
  hi
  I  am  training  QUADRUPLE with 1w person dataset.  I  have  tried  smaller  lr  many  times,   now the acc is about  0.4，but the loss is hard to converge。
   my lr is very small now, only 1e-08. 
my parameters  are  as  follows：
base: 10
gamma: 0.08
power: 1
lambda_min: 10
iteration: 0
    I  have  tried lambda_min：5，but  once  i  changed lambda_min from 10  to  5，the  acc become to 0  immediately。
   what  about  your  acc and  loss  at  last  when  you  trained  with  CASIA.

**Comment by zjz5250** [comment]
@zuoqing1988
  can lamda_min be a large num？for example ：
base：50
lamda_min：50

we tried to train  like this，found the loss can converge to 0.85 for double，with MS_Celeb_1M dataset

**Comment by zuoqing1988** [comment]
@MengWangTHU 
lambda = max(lambda_min,base*(1+gamma*iteration)^(-power))

@zjz5250  larger lambda_min lead to lower accuary, but it is easier to converge.

**Comment by twmht** [comment]
@zuoqing1988 

How do you mean "large" for datasets?

"large" for large number of identities, or for large number of images?
